community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf

```diff
@@ -130,26 +130,22 @@ data "google_compute_zones" "available" {
 }
 
 locals {
-  res_name_split = split("/", var.reservation_name)
-  reservation = var.reservation_name == "" ? null : (
-    length(local.res_name_split) == 4 ? {
-      project : local.res_name_split[1],
-      name : local.res_name_split[3]
-      } : {
-      project : var.project_id,
-      name : var.reservation_name
-    }
-  )
+  res_match = regex("^(?P<whole>(?P<prefix>projects/(?P<project>[a-z0-9-]+)/reservations/)?(?<name>[a-z0-9-]+)(?P<suffix>/[a-z0-9-]+/[a-z0-9-]+)?)?$", var.reservation_name)
+
+  res_short_name = local.res_match.name
+  res_project    = coalesce(local.res_match.project, var.project_id)
+  res_prefix     = coalesce(local.res_match.prefix, "projects/${local.res_project}/reservations/")
+  res_suffix     = local.res_match.suffix == null ? "" : local.res_match.suffix
 
-  reservation_name = local.reservation == null ? "" : "projects/${local.reservation.project}/reservations/${local.reservation.name}"
+  reservation_name = local.res_match.whole == null ? "" : "${local.res_prefix}${local.res_short_name}${local.res_suffix}"
 }
 
 # tflint-ignore: terraform_unused_declarations
 data "google_compute_reservation" "reservation" {
-  count = local.reservation != null ? 1 : 0
+  count = length(local.reservation_name) > 0 ? 1 : 0
```

Comment:
Just curious what the use of `count` here is for, I can see we turn off the tflint for it.

Reply:
Found the answer to my own question [here](https://developer.hashicorp.com/terraform/language/meta-arguments/count)

Reply:
We don't use this `data` anywhere. It's only here for the sake of `lifecycle.postcondition`.